packages/datagateway-dataview/src/views/table/datafileTable.component.tsx

```diff
@@ -161,6 +161,21 @@ const DatafileTable = (props: DatafileTableProps): React.ReactElement => {
     [cartItems, selectAllSetting, allIds]
   );
 
+  const isParentSelected = React.useMemo(() => {
+    return (
+      cartItems?.some(
+        (cartItem) =>
+          cartItem.entityType === 'dataset' &&
+          cartItem.entityId.toString() === datasetId
+      ) ||
+      cartItems?.some(
+        (cartItem) =>
+          cartItem.entityType === 'investigation' &&
+          cartItem.entityId.toString() === investigationId
+      )
```

Comment:
Yeah, we're looping over the `cartItems` array twice here unecessarily - the following code should work equivalently:

```
cartItems?.some((cartItem) => 
  (cartItem.entityType === 'dataset' &&
   cartItem.entityId.toString() === datasetId) ||
  (cartItem.entityType === 'investigation' &&
   cartItem.entityId.toString() === investigationId)
)
```